app/api/packets/helpers.ts

```diff
@@ -315,14 +313,62 @@ export async function getPacket(txHash: string) {
   };
 
   const packetRes = await (await fetch(process.env.INDEXER_URL!, packetOptions)).json();
-  if (packetRes?.data?.packets?.items.length > 0) {
-    packetResponse = packetRes.data.packets.items[0];
-  } else {
-    return [];
-  }
+  return (packetRes?.data?.packets?.items || []);
+}
+
+export async function getRecentPackets(): Promise<[] | [any]> {
+  const headers = {'content-type': 'application/json'};
+
+  const packetRequest = {
+    query: `query Packet {
+              packets(limit:100, orderBy: "sendBlockTimestamp", orderDirection: "desc") {
+                items {
+                  sendPacket {
+                    sequence
+                    sourcePortAddress
+                    sourceChannelId
+                    dispatcherAddress
+                    blockTimestamp
+                    timeoutTimestamp
+                  }
+                  recvPacket {
+                    destPortAddress
+                    destChannelId
+                    blockTimestamp
+                  }
+                  writeAckPacket {
+                    dispatcherAddress
+                    blockTimestamp
+                  }
+                  ackPacket {
+                    blockTimestamp
+                  }
+                  id
+                  state
+                  sendTx
+                  recvTx
+                  writeAckTx
+                  ackTx
+                }
+              }
+            }`
+  };
+
+  const packetOptions = {
+    method: 'POST',
+    headers,
+    body: JSON.stringify(packetRequest)
+  };
+
+  const packetRes = await (await fetch(process.env.INDEXER_URL!, packetOptions)).json();
+  return (packetRes?.data?.packets?.items || []);
+}
+
+export async function processPacket(rawPacket: any): Promise<Packet>{
```

Comment:
It's not good that we have to make a query for each packet to get portId and counterpartyPortid. Let's work on getting that data in one packet's query in the next PR.

Reply:
It will still have to make that channels call for each packet though right? I made it a separate function so I could reuse it on the result of `getPacket(txHash)` and `getRecentPackets()`

Reply:
No need for the channels call if we add the missing information to the packets index